### PR TITLE
Bugfix/543

### PR DESCRIPTION
**Pull request by MMashivskij**
## Description

All components of PDP are rendered corectly.
Similar product for some product are selected by corect algolythm.
The roduct can be added to the cart.

#### Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
image  |  image

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm

### CODE REVIEW

**Comment by Kiyooooko** [comment]
Please  fill checkboxes correctly, please add screenshots
Screenshot 2021-03-11 at 10 53 09

File: src/containers/orders/cart/filled-cart/filled-cart.js [hunk 1]
```diff
@@ -14,32 +16,38 @@ const FilledCart = ({ items }) => {
...
   }));
 
-  const totalPrice = items.reduce(
-    (acc, item) => acc + calcPrice(item, currency),
-    0
-  );
+  const totalPrice = items.reduce((acc, item) => acc + calcPriceForCart(item, currency), 0);
+
+  let currencySign;
+  switch (currency) {
```
Comment (Kiyooooko) [inline src/containers/orders/cart/filled-cart/filled-cart.js:22]: please use ternary operators and move to utils

File: src/pages/product-details/product-details.js [hunk 2]
```diff
@@ -1,67 +1,55 @@
...
   const dispatch = useDispatch();
   const styles = useStyles();
 
   const [sizeIsNotSelectedError, setSizeIsNotSelectedError] = useState(false);
 
-  const currencySign =
-    currency === 0 ? faHryvnia : currency === 1 ? faDollarSign : '';
+  let currencySign;
+
+  switch (currency) {
```
Comment (Kiyooooko) [inline src/pages/product-details/product-details.js:37]: same

File: src/pages/product-details/similar-products/similar-products.js [hunk 3]
```diff
@@ -1,15 +1,64 @@
...
+const SimilarProducts = ({ currencySign, cartList }) => {
   const styles = useStyles();
+  const { language, similarProducts, currency, product } = useSelector(
+    selectInfoForSimilarProducts
+  );
+
+  const { title } = SIMILAR_ITEMS[language];
+
+  const imagesList = similarProducts
+    .filter(
```
Comment (Kiyooooko) [inline src/pages/product-details/similar-products/similar-products.js:24]: please use filter from loads

File: src/pages/product-details/similar-products/similar-products.js [hunk 4]
```diff
@@ -1,15 +1,64 @@
...
+      ({ category, mainMaterial, pattern }) =>
+        category._id !== product.category._id &&
+        (mainMaterial.color._id === product.mainMaterial.color._id ||
+          pattern._id === product.pattern._id)
+    )
+    .map(({ _id, images, rate, name, basePrice }) => (
```
Comment (Kiyooooko) [inline src/pages/product-details/similar-products/similar-products.js:30]: please use map from lodash

File: src/redux/products/products.operations.js [hunk 5]
```diff
@@ -24,12 +24,12 @@ const getAllFilters = async () => {
...
-            name {
-              value
-            }
-          }
+          # closures {
+          #   _id
+          #   name {
+          #     value
+          #   }
+          # }
```
Comment (Kiyooooko) [inline src/redux/products/products.operations.js:32]: why you need this from line 27 to 32?

**Comment by Kiyooooko** [comment]
When size is default, I can't add to card product
Screenshot 2021-03-11 at 11 15 11

**Comment by Kiyooooko** [comment]
We can't plus or minus product in cart
Screenshot 2021-03-11 at 11 16 17

**Comment by Kiyooooko** [comment]
We can't delete products from cart
Screenshot 2021-03-11 at 11 23 03

**Comment by Kiyooooko** [comment]
We can't go back to product list from product details
Screenshot 2021-03-11 at 11 23 51

**Comment by Kiyooooko** [comment]
We can't add more than 5 products to cart
Screenshot 2021-03-11 at 11 25 11

**Comment by Kiyooooko** [comment]
Delivery type is empty when we choose it
Screenshot 2021-03-11 at 11 27 03

**Comment by Kiyooooko** [comment]
Confirm order don't work
Screenshot 2021-03-11 at 11 28 59

**Comment by Kiyooooko** [comment]
When we go to cart from main page, there is an error
Screenshot 2021-03-11 at 11 30 02

**Comment by Kiyooooko** [comment]
please make "git pull origin development"

File: src/containers/orders/cart/filled-cart/filled-cart.js (hunk 1 above)
Comment (MMashivskij) [reply src/containers/orders/cart/filled-cart/filled-cart.js:22]: CI said, that ternary operator in ternary operator, is code smell, that why I use "switch".

File: src/pages/product-details/product-details.js (hunk 2 above)
Comment (MMashivskij) [reply src/pages/product-details/product-details.js:37]: CI said, that ternary operator in ternary operator, is code smell, that why I use "switch".

File: src/containers/orders/cart/filled-cart/filled-cart.js (hunk 1 above)
Comment (Kiyooooko) [reply src/containers/orders/cart/filled-cart/filled-cart.js:21]: you should use ternary and move to utils

File: src/pages/product-details/product-details.js (hunk 2 above)
Comment (Kiyooooko) [reply src/pages/product-details/product-details.js:37]: you should use ternary and move to utils!

**Comment by Kiyooooko** [comment]
Fix that please
Screenshot 2021-03-11 at 12 23 12

File: src/utils/currency.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export const currencySign = (currency, faHryvnia, faDollarSign) => currency === 0 ? faHryvnia : currency === 1 ? faDollarSign : '';
```
Comment (Kiyooooko) [inline src/utils/currency.js:1]: you should do two ternary, because you have ternary in ternary

**Comment by M1ndForce** [comment]
if a product already exists in the cart should display an error on the PDP page "this product already in cart"
image

**Comment by M1ndForce** [comment]
Please style this button to our project style guide
image

**Comment by M1ndForce** [comment]
When changing quantity total price doesn't change
image

**Comment by M1ndForce** [comment]
When press buy now you cant make an order for this product
image

**Review by Haytham1337 — DISMISSED** [review]

**Review by AndriyLymych — APPROVED** [review]

**Review by Haytham1337 — APPROVED** [review]

**Review by M1ndForce — APPROVED** [review]

**Pull request merged**